index.js

```diff
@@ -72,3 +132,11 @@ function isAllowPublic (input) {
 
   return true
 }
+
+function isAllowPublic2 (input) {
```

Comment:
:fire: this function should only be one line:

```js
  return get(input, ['options', 'allowPublic']) === true
```

The other things are different concerns?
1. you're checking if it's already encrypted? (content not an object) You already checked that
2. you're checking if type is present and a string? No idea why?

Reply:
i just copied this from `isAllowPublic`, it looked important. 

it is strange that the

```js
        if (isAllowPublic2(input)) {
          if (hasRecps(input.content)) {
```

check comes after the publish

huh now i found a bug in the publish hook. `hasRecps` gets sent `input` as an arg but `hasRecps` is supposed to take `content` as an arg. the `      if (isAllowPublic(input)) {` block in the publish hook should come first, but it comes second. the only reason it's reached is because the first `if` block misses the recps

Reply:
ok idk if that was a bug but i simplified the code a bunch and also deduplicated it so now there's only 1 hook function!